### DISCUSSION TITLE
How to use raw packet data

### DISCUSSION THREAD

**Discussion by MuddyRock** [Arnold-n/P1P2MQTT#153] [General]
Hi
I'm trying to use MQTT to change a setting on my Daikin E-series (Set Water pipe freeze protection to Off).
I set the bridge to mode J1 to send raw packets to P1P2/R/P1P2MQTT/bridge0 then changed the setting and then set the mode to J0.

The packets are listed below.

How do I find which packet changes the setting and then use it in MQTT publish? I've tried the most likely (any line with 35 or 36 for packet type in 3rd byte) and sent the payload via MQTT publish e.g. payload of E3611004200FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF6C but the bridge errors (monitored on telnet) with 

``
wr_nr[0]: 0x1100 >= nr_params : 0x2F 
``

If I try a line without the right packet type, I get 

``
wr_pt[0]: 0x0 not 0x35-0x3D
``

Any help much appreciated :)

Edit: attempt to format packet block as code.

``
R 2024-12-13 15:10:37 T  0.025: 000015000064000064000000000000000000000000000007
R 2024-12-13 15:10:37 T  0.024: 40001500000700000002FFFF007D2B08C8C8D7600A3800FC
R 2024-12-13 15:10:37 T  0.025: 00001600007F280000000000000000000000000000000055
R 2024-12-13 15:10:37 T  0.024: 400016000000000000000000000000000000000000000036
R 2024-12-13 15:10:37 T  0.025: 00F130010100000A0300000701010000004E
R 2024-12-13 15:10:37 T  0.179: 000080010200080002000804150E080104000842020008A1
R 2024-12-13 15:10:37 T  0.024: 400080C1020008C0020008C4150E08C1040008C202000843
R 2024-12-13 15:10:37 T  0.025: 00FF30010100000F0500010A0101010017E8
R 2024-12-13 15:10:37 T  0.179: 00F0300100000000010000000000000000C0
R 2024-12-13 15:10:38 T  0.100: 40F0300000000000000000000000000000AD
R 2024-12-13 15:10:38 T  0.025: 00F031000000096000180C0D0E093654
R 2024-12-13 15:10:38 T  0.030: 40F031000000087010180C0D0E09369B

R 2024-12-13 15:10:38 T  0.025: 00F03611004200FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF6C

R 2024-12-13 15:10:38 T  0.030: 40F036FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF7F
R 2024-12-13 15:10:38 T  0.436: 0000100001000000000014000000000800001400001E005D
R 2024-12-13 15:10:38 T  0.024: 400010000001001E00140014001A00000000000000000022
R 2024-12-13 15:10:38 T  0.025: 000011140100000000000000000000000000000000000034
R 2024-12-13 15:10:38 T  0.024: 40001180012DB807000AF40A380A08140106880000000060
R 2024-12-13 15:10:38 T  å0.025: 00001200040E09180C0D1900C8000040140000000000007B
R 2024-12-13 15:10:38 T  0.024: 4000124000000000000000FE04007F010400000000000031
R 2024-12-13 15:10:38 T  0.025: 000013000050000000000000000000000000000000000013
R 2024-12-13 15:10:39 T  0.024: 4000131E000B5080800A8000000772FFFF0000201000001F
R 2024-12-13 15:10:39 T  0.025: 00001426001400190014000A00000000000000000000002B
R 2024-12-13 15:10:39 T  0.024: 40001426001400190014000A0000000000002B002300006C
R 2024-12-13 15:10:39 T  0.025: 000015000064000064000000000000000000000000000007
R 2024-12-13 15:10:39 T  0.024: 40001500000700000003FFFF007D2B08C8C8D76E0A3800F6
R 2024-12-13 15:10:39 T  0.025: 00001600007F280000000000000000000000000000000055
R 2024-12-13 15:10:39 T  0.024: 400016000000000000000000000000000000000000000036
R 2024-12-13 15:10:39 T  0.025: 00FF30010100000F0500010A0101010017E8
R 2024-12-13 15:10:39 T  0.179: 00F0300100000000010000000000000000C0
R 2024-12-13 15:10:39 T  0.100: 40F0300000000000000000000000000000AD
R 2024-12-13 15:10:39 T  0.025: 00F031000000096000180C0D0E093784
R 2024-12-13 15:10:39 T  0.030: 40F031000000087010180C0D0E09374B
R 2024-12-13 15:10:39 T  0.025: 00F0360500C90111004100FFFFFFFFFFFFFFFFFFFFFFFF01
R 2024-12-13 15:10:39 T  0.030: 40F036FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF7F
R 2024-12-13 15:10:39 T  0.025: 00F130010100000A0300000701010000004E
``

**Comment by Arnold-n** [comment]
This seems something easier done (once) via the Daikin controller, as field settings are not intended for frequent changes: both risky to have freeze protection off, and any field setting change requires a Daikin system restart.

The water freeze protection change is visible in the last 4 bytes of the following packet above:
`000080 01020008 00020008 04150E08 01040008 42020008`.

The P1/P2 protocol is not designed to just insert a new message. To change field setting [4-04], you can give the following command:
-`E390040080002XX` where XX=00 for continuous pump operation, XX=01 for non-continues pump operation, or XX=02 for OFF.
-followed by a restart `L99` or button press `Daikin_Restart_Careful` in HA if the compressor is off.
Also, `J0` silences the bridge, to get back to default operation, you would want to use `J2`.

**Reply by MuddyRock** [reply]
Thankyou - I will try this later. I set it to J0 temporarily so I could capture the packets, but I did remember to put it back to J2 for normal use.
